pkg/scheduling/v2/scheduler.go

```diff
@@ -735,9 +743,13 @@ func (s *Scheduler) tryAssign(
 					for _, singleRes := range results {
 						if !singleRes.succeeded {
 							if singleRes.rateLimitResult != nil {
-								rateLimited = append(rateLimited, singleRes.rateLimitResult)
-							} else if singleRes.noSlots {
-								unassigned = append(unassigned, singleRes.qi)
+								batchRateLimited = append(batchRateLimited, singleRes.rateLimitResult)
+							} else {
+								batchUnassigned = append(batchUnassigned, singleRes.qi)
+
+								if !singleRes.noSlots {
```

Comment:
This case shouldn't happen, and as far as I can tell hasn't happened, but it's good just to double check.